building_map_tools/building_crowdsim/test/build_configfile_TEST.py

```diff
@@ -0,0 +1,131 @@
+import os
+
+import xml.etree.ElementTree as ET
+
+from building_crowdsim.config.scene_file import\
+    SceneFile, ObstacleSet, AgentProfile, AgentGroup
+from building_crowdsim.config.behavior_file import\
+    BehaviorFile, BehaviorState, StateTransition, GoalSet
+from building_crowdsim.config.plugin_file import Plugin
+import building_crowdsim.config.util as util
+from building_crowdsim.building_yaml_parse import BuildingYamlParse
+
+
+def simple_unit_test():
```

Comment:
First of all I would rename the test to be a bit more explicative, also I don't quite understand what the tests are supposed to check since there is no checking of the generated files. Do the only test that the functions managed to run without triggering one of the raise / asserts?

Reply:
For the configfile generation, I guess I put lots of raise ValueErro in the code, so that if the test function successfully ends without any ValueError, the test should be fine. When dealing with Navmesh, even though the function ends without any assertion and ValueError, the navmeh itself might be inappropriate in menge's logic due to some parameter selection. The base_test provides a simple test case full of special cases so that we can check whether the generated file is correct. As for the configfile, it actually just transforms the yaml configuration to the xml format under the menge definition. With the raise and is_valid function defined in each type, I think if the test function successfully ends the test should be fine. Although I don't know whether put checks inside the code is the proper way to do it.